### ISSUE TITLE
Option to hide Act/Act Items from list?

### ISSUE THREAD

**Issue by PatriciaMoller** [SmartZoos/Toolset#192]
If a Activity has been created for a specific purpose where they would want others to see (teacher created one for a test, staff has an event planned and don't want to publish it too early), would it be possible to hide it from the public Actvity list. But still have it sharable via URL/QR.

**Comment by pjotrsavitski** [comment]
Yes, but we would need a special status for that that could be set.

Something like: "Shown in listings"

If I understood the requirement correctly, then it is needed for that Activity to be fully functioning and available, just excluded from the public listings. The same way YouTube does with uploaded videos. If someone knows the URL (even if that is just discovered by an accident), then the Activity would still be playable.

That would probably have to be a checkbox that is part of the create/edit form. Then it would also have to be respected by the listing and search code logic. Not a big deal, but do tell me if I got it right.

**Comment by PatriciaMoller** [comment]
Yes, that sounds right!

**Comment by pjotrsavitski** [comment]
I have though about this one. The listings are also used by admins, creators and others. If something gets hidden, then how could it be found. Example: someone create and Activity, marks that as hidden, then whats to edit that, but as it is hidden from the listing it is nowhere to be found. What would be the approach to deal with that?

Hiding Activity Items from the list could have more consequences. Should it also affect the part when Activity is being created (search for items)? What should be done with items that have already been attached to the activities?

The lists could show different things for different users, based on role or ownership. Not sure if that could create confusion or not. There could also be something like: My Activities or My Activity Items.  That one would show all the things created by currently logged in user.

**Comment by kaimikael** [comment]
It seems reasonable that a creator always can see their own creations, but others don’t necessarily do so.  The default would be that all can see an item, but that it can be turned off.  Then when it’s turned on, it can be accessed by others.  Having a simple access control permission seems to me easier than having to hand around URLs.  Though, you could imagine a small group of people wanting to be able to show each other stuff before they are made public, in which case you’d need to introduce the idea of groups and group membership…

**Comment by pjotrsavitski** [comment]
Well, this issue was about hiding from the lists. Not tuning the permission system. That could also be done, but there does not seem to be much point in creating groups that are able to access certain activities.

We already have special roles that should be able to edit/delete based on certain conditions. That could be extended to being able to view certain items. Although, that is a rather different issue.

This would also affect the Share via QUR code functionality as that would fails to work.

**Comment by PatriciaMoller** [comment]
@kaimikael How would another person see the activity after you have given access? Would they be notified  somehow?

**Comment by kaimikael** [comment]
I have assumed the activities they have gotten permission to see will be visible when they next look at the list, so no particular notification.
